### ISSUE TITLE
Feature Enhancement suggestions

### ISSUE THREAD

**Issue by chaz9456** [jakob/Postico#270]
These may exist already, but I couldn't find where this setting would be enabled in the app.

The best part of postico is that it rarely crashes and it's very lightweight. I migrated from heavy pgadmin iii use because this provides a really slick interface and I can finally escape the pgadmin headaches and tableflipping with it's unstable query editor and frequent crashes. 

However, what I think postico really needs is 1. parenthetical highlighting and 2. clearer debug messages. pgadmin iii has both of these and when working on debugging a complex SELECT query in the editor I find myself copying and pasting into pgadmin query editor to figure out if I have unbalanced parentheses and what position of the query is really causing issues (and why).

In short, parenthetical highlighting in the editor would go a long way in beefing up this app. I love doing scratch work and experimenting and exploring in postico, but this missing feature sucks. when the queries become sufficiently complex, it's good to be able to click the cursor at a certain parenthesis and have a visually indication in the query window which other parenthesis it is paired with it. Pgadmin does a great job with this and it's a feature that is oft taken for granted, but once it's gone you really wish you had it.

On a more minor note, the debug messages leave more to be desired. But this would be less of an issue with good query editor syntactic highlights for parenthetical imbalances.

**Comment by jakob** [comment]
Thanks for your suggestion! It looks like you've suggested parenthetical highlighting before (#199). I agree that this would be very useful.

Do you have more specific suggestions concerning error messages? Most of them come directly from the server, but if you have specific ideas how to improve them, I'd be glad to hear them.

**Comment by chaz9456** [comment]
No problem. Always glad to provide input. this is definitely one of the best postgres clients i've used.

Sorry for the slow reply. 

I should have been less confusing regarding my debug message comment. What I like is that pgadmin III highlights the exact spot of the error so that it can be quickly found. it does this via a green squiggly underscore. with large queries it makes it easier to find than just a simple "which line did it occur in" from the debug dialog. Of course pgadmin doesn't have line numbers and postico does (another win for postico), but the squiggly highlighting I think would be another added benefit. again note the "more minor note", this may be something that runs up against returns to effort.

Thanks for listening to my comments.

**Comment by jakob** [comment]
Just a short update: We've implemented the part with the matching parentheses in [Postico 2](#689)
